daemon/resize.go

```diff
@@ -37,5 +37,7 @@ func (daemon *Daemon) ContainerExecResize(name string, height, width int) error
 	if err != nil {
 		return err
 	}
+	ec.Lock()
```

Comment:
Could you add a comment here how this lock is correcting the synchronization between request handlers? What request is holding the lock and keeps this waiting etc.

Reply:
I add some comments in a new commit, please help take a look. 
Thx~!